saleor/events/models/OrderEvent.py

```diff
@@ -0,0 +1,274 @@
+        return 'OrderEvent(type=%r, user=%r)' % (self.type, self.user)
+
+    @staticmethod
+    def _lines_per_quantity_to_str_line_list(quantities_per_order_line):
+        return [{
+            'quantity': quantity,
+            'item': str(line)
+        } for quantity, line in quantities_per_order_line]
+
+    def get_event_display(self):
+        return display_order_event(self)
+
+    @classmethod
+    def email_sent_event(
+            cls, *,
+            order: Order, email_type: OrderEventsEmails,
+            source: Optional[User]) -> models.Model:
+
+        if source is not None and source.is_anonymous:
+            source = None
+
+        return cls(
+            order=order, type=OrderEvents.EMAIL_SENT.value,
+            user=source,
+            parameters={
+                'email': order.get_user_current_email(),
+                'email_type': email_type.value})
+
+    @classmethod
+    def email_resent_event(
+            cls, *,
+            order: Order, email_type: OrderEventsEmails,
+            source: User) -> models.Model:
+        raise NotImplementedError
+
+    @classmethod
+    def draft_created_event(
+            cls, *, order: Order, source: User) -> models.Model:
+        return cls(
+            order=order, type=OrderEvents.DRAFT_CREATED.value, user=source)
+
+    @classmethod
+    def draft_added_products(
+            cls, *,
+            order: Order, source: User,
+            order_lines: List[Tuple[int, ProductVariant]]) -> models.Model:
+
+        return cls(
+            order=order, type=OrderEvents.DRAFT_ADDED_PRODUCTS.value,
+            user=source,
+            parameters={
+                'lines': cls._lines_per_quantity_to_str_line_list(order_lines)
+            })
+
+    @classmethod
+    def draft_removed_products(
+            cls, *,
+            order: Order, source: User,
+            order_lines: List[Tuple[int, ProductVariant]]
+    ) -> models.Model:
+
+        return cls(
+            order=order, type=OrderEvents.DRAFT_REMOVED_PRODUCTS.value,
+            user=source,
+            parameters={
+                'lines': cls._lines_per_quantity_to_str_line_list(
+                    order_lines)})
+
+    @classmethod
+    def placed_event(
+            cls, order: Order, source: User, from_draft=False) -> models.Model:
+        event_type = (
+            OrderEvents.PLACED_FROM_DRAFT if from_draft else OrderEvents.PLACED
+        )
+
+        if source.is_anonymous:
+            source = None
+
+        return cls(
+            order=order, type=event_type.value, user=source)
+
+    @classmethod
+    def draft_oversold_items_event(
+            cls, *,
+            order: Order, source: User,
+            oversold_items: List[str]) -> models.Model:
+        return cls(
+            order=order, type=OrderEvents.OVERSOLD_ITEMS.value,
+            user=source,
+            parameters={
+                'oversold_items': oversold_items})
+
+    @classmethod
+    def cancelled_event(
```

Comment:
We once discussed whether to use British or American spelling in Saleor, and we agreed on American. In American spelling this should be "canceled_event", also the event type inside is called `CANCELED`.